lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -513,6 +523,39 @@ fromNetworkMagic :: NetworkMagic -> W.ProtocolMagic
 fromNetworkMagic (NetworkMagic magic) =
     W.ProtocolMagic (fromIntegral magic)
 
+--
+-- Stake pools
+--
+
+fromPoolId :: SL.KeyHash 'SL.StakePool crypto -> W.PoolId
+fromPoolId (SL.KeyHash x) = W.PoolId $ getHash x
+
+fromPoolDistr
+    :: SL.PoolDistr TPraosStandardCrypto
+    -> Map W.PoolId Percentage
+fromPoolDistr =
+    Map.map (unsafeMkPercentage . fst)
+    . Map.mapKeys fromPoolId
+    . SL.unPoolDistr
+
+-- TODO: Change to return a map of maps, instead of using head
+fromRewards
+    :: O.NonMyopicMemberRewards TPraosStandardCrypto
+    -> Map W.PoolId (Quantity "lovelace" Word64)
+fromRewards =
+    Map.map (Quantity . fromIntegral)
+    . Map.mapKeys fromPoolId
+    . snd
+    . head
+    . Map.toList
+    . O.unNonMyopicMemberRewards
+
+optimumNumberOfPools :: SL.PParams -> Int
+optimumNumberOfPools = safeConvert . SL._nOpt
+  where
+    safeConvert :: Natural -> Int
+    safeConvert = fromIntegral
```

Comment:
Is it though :wink: 

```
λ> safeConvert 9223372036854775808
-9223372036854775808
```

Reply:
I don't think going from an unbounded type to a bounded type will ever be "safe". Although realistic values of _optinumNumberOfPools_ are most likely within bounds.

Reply:
Yes, this should be labeled "unsafeConvert" (perhaps with a comment). Not sure how I managed to get to this.

Reply:
5f93fcd9cee368ea11069a0337051f8894558b45